### ISSUE TITLE
[question] Package lists, promotion and virtual remotes

### ISSUE THREAD

**Issue by antoinebardoz** [conan-io/conan-extensions#136]
### What is your question?

I'm trying to implement package promotion using package lists, but I'm having trouble with using a virtual remote as the source remote.

I create the package list like this:
```
conan install --require=top_package/version@company -r conan-virtual -f json > create_dev.json
conan list --graph=create_dev.json --format=json > pkglist_dev.json
```

My extension uses the REST API, with URL's such as:
```
 /artifactory/api/copy/conan-virtual/company/package/2.0.0/_/a9e11b8366f22f0e50ac3dcff74da3c4/export?to=/conan-prod/company/package/2.0.0/_/a9e11b8366f22f0e50ac3dcff74da3c4/export&suppressLayouts=1&dry=1
```

This leads to a 409 error. If the source remote is _not_ virtual, there is no problem.

This is a fairly severe limitation, and I need some help dealing with this. Either of the following would be an acceptable solution to me:
1. A way to determine which non-virtual remote the package is actually in.
or, if necessary:
2. Check if a remote is virtual, so I can give a sensible error message

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
I have got some further insights:
- The Artifactory API copy only supports "Local, remote and federated repositories", see https://jfrog.com/help/r/jfrog-rest-apis/copy-item
- So this seems by design of the Artifactory API

I have gotten some feedback, and trying to get the "local" repository of every different artifact that a "virtual" repository is aggregating doesn't seem like a great approach. 

So the other possibility would be to at least know the repository types. This seems to be part of the rest API, like https://jfrog.com/help/r/jfrog-rest-apis/get-repositories, seems doable.

To summarize, it seems that copy promotions are designed and intended to be ran from a local repo, and that local repo should be known and explicitly addressed. Using Conan you might run a ``conan download + conan upload`` client side promotion, but that would be much slower and expensive in time and money, so probably not recommended in most cases.

**Comment by antoinebardoz** [comment]
Thanks, I've been able to use the 'repositories' command to get the remote type. This definitely helps!
